lib/index.js

```diff
@@ -100,15 +108,26 @@ internals.implementation = function (server, options) {
 
                 request.auth.artifacts = null;
                 reply.unstate(settings.cookie);
-            },
-            ttl: function (msecs) {
+            };
 
+            this.ttl = function (msecs) {
+
+                const reply = self.reply;
                 const session = request.auth.artifacts;
                 Hoek.assert(session, 'No active session to modify ttl on');
                 reply.state(settings.cookie, session, { ttl: msecs });
-            }
+            };
         };
 
+        return new cookieAuth();
+    };
+
+    server.decorate('request', 'cookieAuth', decoration, { apply: true });
+
+    server.ext('onPreAuth', (request, reply) => {
+
+        request.cookieAuth.reply = reply;
```

Comment:
I would add a comment that reply is only used for setting and unsetting state, and not for actual replies.